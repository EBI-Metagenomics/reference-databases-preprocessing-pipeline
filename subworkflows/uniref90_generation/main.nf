include { UNIREF90_RHEA_FILTER                  } from '../../modules/uniref90_rhea_filter/main.nf'
include { UNIREF90_NON_VIRAL_FILTER             } from '../../modules/uniref90_non_viral_filter/main.nf'
include { REFORMAT_RHEA_CHEBI                   } from '../../modules/reformat_rhea_chebi/main.nf'
include { DIAMOND_MAKEDB as DIAMOND_MAKEDB_RHEA } from '../../modules/nf-core/diamond/makedb/main.nf'
include { DIAMOND_MAKEDB as DIAMOND_MAKEDB_TAXA } from '../../modules/nf-core/diamond/makedb/main.nf'

workflow UNIREF90_GENERATION {

    main:
        uniref90_fasta       = file(params.uniref90_download_fasta, checkIfExists: true)
        uniprot_rhea_mapping = file(params.uniprot_rhea_mapping, checkIfExists: true)
        rhea_chebi_mapping   = file(params.rhea_chebi_download_mapping, checkIfExists: true)

        UNIREF90_RHEA_FILTER(
            uniref90_fasta,
            uniprot_rhea_mapping
        )

        UNIREF90_RHEA_FILTER.out.filtered_fasta
            .map { filepath -> 
                [[id: params.uniref90_version], filepath]
            }
            .set { diamond_makedb_rhea_ch }
        DIAMOND_MAKEDB_RHEA(diamond_makedb_rhea_ch, [], [], [])

        uniref90_batches_ch = Channel.fromPath(uniref90_fasta).splitFasta(by: 1000000, file: 'uniref90')
        UNIREF90_NON_VIRAL_FILTER(
            uniref90_batches_ch,
        )

        UNIREF90_NON_VIRAL_FILTER.out.filtered_fasta
            .collectFile(name: 'uniref90_non_viral.fasta')
            .map { filepath ->
                [[id: params.uniref90_version], filepath]
            }
            .set { diamond_makedb_taxa_ch }
        DIAMOND_MAKEDB_TAXA(diamond_makedb_taxa_ch, [], [], [])

        REFORMAT_RHEA_CHEBI(
            rhea_chebi_mapping
        )

    emit:
        rhea_db            = DIAMOND_MAKEDB_RHEA.out.db
        taxonomy_db        = DIAMOND_MAKEDB_TAXA.out.db
        rhea_chebi_mapping = REFORMAT_RHEA_CHEBI.out.tsv_rhea_chebi_mapping
}
