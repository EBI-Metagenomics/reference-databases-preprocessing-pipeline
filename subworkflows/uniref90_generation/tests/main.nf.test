nextflow_workflow {

    name "Test workflow UNIREF90_GENERATION"
    script "../main.nf"
    workflow "UNIREF90_GENERATION"

    test("Run the workflow on the test dataset") {

        when {
            workflow {

            }
            params {
                    uniref90_download_fasta     = "${moduleDir}/tests/data/uniref90.test.fasta"
                    uniprot_rhea_mapping        = "${moduleDir}/tests/data/uniprot_rhea_mapping.test.tsv"
                    rhea_chebi_download_mapping = "${moduleDir}/tests/data/rhea-reactions.txt.gz"
            }
        }

        then {
            assertAll(
            { assert workflow.success },
            { assert snapshot(workflow.out.rhea_db).match("rhea_db") },
            { assert snapshot(workflow.out.taxonomy_db).match("taxonomy_db") },
            { assert snapshot(workflow.out.rhea_chebi_mapping).match("rhea_chebi_mapping") }
            )
        }
    }
}
