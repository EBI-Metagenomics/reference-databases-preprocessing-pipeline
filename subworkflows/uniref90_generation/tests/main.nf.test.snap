{
    "rhea_db": {
        "content": [
            [
                [
                    {
                        "id": "uniref90_rhea_2024_05_2024-07-31"
                    },
                    "uniref90_rhea_2024_05_2024-07-31.dmnd:md5,24d32b7dfd8efa693ddb182d34465070"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-19T15:56:15.881724"
    },
    "taxonomy_db": {
        "content": [
            [
                [
                    {
                        "id": "uniref90_taxa_2024_05"
                    },
                    "uniref90_taxa_2024_05.dmnd:md5,d0d7be70ac6d182ba5da21602b3aa566"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T13:35:37.731581"
    },
    "rhea_chebi_mapping": {
        "content": [
            [
                "rhea_chebi_mapping_135.tsv:md5,52d55d130ac9e9329e6e32d88b335f58"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-18T15:15:49.259818"
    }
}